src/django_mysql/models/lookups.py

```diff
@@ -69,6 +70,25 @@ class JSONExact(JSONLookupMixin, Exact):
     pass
 
 
+class JSONIn(In):
+    lookup_name = "in"
+
+    def get_prep_lookup(self):
+        if hasattr(self.rhs, "_prepare"):
+            return super(JSONIn, self).get_prep_lookup()
+        prepared_values = []
+        for rhs_value in self.rhs:
+            if hasattr(rhs_value, "resolve_expression"):
+                pass
+            else:
+                if isinstance(rhs_value, (dict, list, tuple)):
+                    rhs_value = JSONValue(rhs_value)
+                else:
+                    rhs_value = Value(json.dumps(rhs_value, allow_nan=False))
```

Comment:
same here as I comented on #621 - might be better to whitelist types that can go through `Value`. Also it would be better to use a `Value` subclass rather than `json.dumps` directly here - perhaps rename the original `CastJSONValue` and this one `JSONValue`?

Reply:
renaming and changelog done, @adamchainz

Reply:
and PR description edited